### PR TITLE
Helm integration tests for release branches

### PR DESCRIPTION
**Pull request by chillleader**
## Description

Follow up on https://github.com/camunda/connectors/pull/2530.

Previously, INTEGRATION_TEST.md would always pass in the default values. This works well for SNAPSHOT's.

However, for patches on minor versions (especially 8.3), the Helm integration tests need to be passed in dependencies that correspond to the that minor version.

This PR will:

 - determine the correct connectors version to use based on the maven project version (if not provided)
 - determine the correct Helm chart to use (i.e. Helm chart repository git ref)
 - call the integration-test-template.yml with these inputs

ℹ️ Compared to the PR from @ev-codes (#2530), this approach should be simpler because it assumes that we don't need to provide the whole values file in `extraValues` when calling a custom Helm charts branch.

## Related issues

closes https://github.com/camunda/team-connectors/issues/764

### CODE REVIEW

File: .github/workflows/INTEGRATION_TEST.yml [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
+    branches:
+      - release/*
   workflow_dispatch:
     inputs:
-      connectors_version:
-        description: 'The version of the connectors to test'
-        required: true
-        default: 'SNAPSHOT'
+      connectors-version:
+        description: 'The version of the connectors to test. If not provided, the version is determined based on the Maven project version. On main branch, the default is SNAPSHOT'
```
Comment (igpetrov) [inline .github/workflows/INTEGRATION_TEST.yml:11]: Suggestion: could also apply some 'branding'

```suggestion
        description: 'The version of the Connectors to test. If not provided, the version is determined based on the Maven project version. On main branch, the default is SNAPSHOT'
```

**Review by igpetrov — DISMISSED** [review]
LGTM

**Review by igpetrov — APPROVED** [review]

**Comment by chillleader** [comment]
I'm moving this back to draft state because the Helm integration test GHA will undergo major refactoring and the inputs will change next week. [🔒 Source](https://camunda.slack.com/archives/C06BN4SRB6C/p1716460112152469?thread_ts=1716448671.646989&cid=C06BN4SRB6C)

**Comment by chillleader** [comment]
We're merging this now to benefit from this solution already while the workflow refactoring on the Distro side is still in progress

**Pull request merged**